### ISSUE TITLE
Add median absolute deviation and standard deviation to `skimage.filter.rank`

### ISSUE THREAD

**Issue by AlvarHHM** [scikit-image/scikit-image#2806]
Or is there a way to extend the functionality without changing the package (just like in scikit-learn, you can extend the BaseEstimator to create a new classifier)

**Comment by jni** [comment]
Hi @AlvarHHM!

The `filters.rank` filters are based on a complicated moving-histogram implementation that is quite difficult to extend. However, you can create new filters using `scipy.ndimage.generic_filter`. Have a look at my blog post for how to do it efficiently:

https://ilovesymposia.com/2017/03/15/prettier-lowlevelcallables-with-numba-jit-and-decorators/

You might want to follow the links to the previous posts for a bit of background.

**Comment by stefanv** [comment]
Note that rank filters do support custom kernels: https://github.com/scikit-image/scikit-image/blob/master/skimage/filters/rank/generic_cy.pyx#L40

**Comment by AlvarHHM** [comment]
Is there a simple way that I can add a custom kernel without altering the source code in the package?

**Comment by stefanv** [comment]
I don't think so, at least not at the moment.  For that, you'll be best served by numba, as @jni outlined above.

**Comment by hsuominen** [comment]
Implementing standard deviation at a minimum would indeed be extremely helpful
